fontir/src/variations.rs

```diff
@@ -243,6 +286,41 @@ impl VariationModel {
 
         Ok(result)
     }
+
+    /// Convert relative deltas to absolute values at the given location.
+    ///
+    /// Rust version of <https://github.com/fonttools/fonttools/blob/4ad6b0db/Lib/fontTools/varLib/models.py#L514-L545>
+    pub fn interpolate_from_deltas<V>(
+        &self,
+        location: &NormalizedLocation,
+        deltasets: &[(VariationRegion, Vec<V>)],
+    ) -> Vec<V>
+    where
+        V: Copy + Mul<f64, Output = V> + Add<V, Output = V>,
+    {
+        let mut result = None;
+        for (region, deltas) in deltasets.iter() {
+            let scalar = region.scalar_at(location).into_inner() as f64;
+            if scalar == 0.0 {
+                continue;
+            }
+            let contribution = deltas.iter().map(|d| *d * scalar).collect::<Vec<_>>();
```

Comment:
taking this as written, there's no need to collect here; you only need to actually allocate a new vec if the result is currently none.